### PR TITLE
.gitignore, rnaseq.rmd dependencies, rseqc read_distribution

### PR DESCRIPTION
**Pull request by njohnso6**
Issues #223 and #232 
The dependency thing should be tested to see if the functional enrichment cache is too easily invalidated. The parent objects it relies on ma y be be changed in previous chunks but the attributes themselves may not be. The documentation is unclear at what level that operates. IT appears it may be conservative at the level of the object. But it is not clear.

### CODE REVIEW

File: workflows/rnaseq/Snakefile [hunk 1]
```diff
@@ -760,6 +760,20 @@ rule rseqc_infer_experiment:
...
+    """
+    read distribution plots
+    """
+    input:
+        bam=c.patterns['bam'],
+        bed12=c.refdict[c.organism][config['gtf']['tag']]['bed12'],
+    output:
+        txt=c.patterns['rseqc']['read_distribution']
+    log:
+        c.patterns['rseqc']['infer_experiment'] + '.log'
```
Comment (daler) [inline workflows/rnaseq/Snakefile]: `s/infer_experiment/read_distribution/g`

File: workflows/rnaseq/downstream/rnaseq.Rmd [hunk 2]
```diff
@@ -29,7 +29,10 @@ output:
 ```
 
 ```{r global_options, include=FALSE}
-knitr::opts_chunk$set(warning=FALSE, message=FALSE)
+knitr::opts_chunk$set(warning=FALSE, message=FALSE, autodep=TRUE)
+
+#automatically identify cache dependencies
+knitr::dep_auto()
```
Comment (daler) [inline workflows/rnaseq/downstream/rnaseq.Rmd]: whoa . . . have you had a chance to test this in practice?

Reading through https://yihui.name/knitr/demo/cache/ to learn more about other things in the cache I've missed, I saw https://github.com/yihui/knitr/issues/238 which uses the md5sum of an input data file. 

What are your thoughts on using the md5sum of the featurecounts file for this?

Comment (njohnso6) [reply workflows/rnaseq/downstream/rnaseq.Rmd]: It takes about .03 seconds for a 5 MB file, not a bad idea. 
Perhaps also the sampletable?
I have not had a chance to test this in practice using our data and variables. However, I have tested it with our code and dummy variables. I'll test it a little now to see how it fares.

**Comment by daler** [comment]
Closing in favor of #241 which is mostly a duplicate of this one

**Pull request closed without merging**